pkg/operator/controller/externaldns/deployment.go

```diff
@@ -320,14 +320,14 @@ func externalDNSDeploymentChanged(current, expected *appsv1.Deployment) (bool, *
 // externalDNSAnnotationsChanged returns true if the current secret annotation differ from the expected
 func externalDNSAnnotationsChanged(current, expected, updated *appsv1.Deployment) bool {
 	changed := false
-	if current.Annotations == nil {
-		updated.Annotations = expected.Annotations
+	if current.Spec.Template.Annotations == nil {
+		updated.Spec.Template.Annotations = expected.Annotations
```

Comment:
Template Spec annotations are set from the expected deployment annotations while they have to be set from the expected template spec annotations.